platform/admin/core/admin-core-alertservice/src/main/java/org/codice/ddf/admin/core/alert/service/internal/AlertServiceBean.java

```diff
@@ -0,0 +1,202 @@
+
+    private final MBeanServer mBeanServer;
+
+    private final InsecureDefaultsService insecureDefaultsService = new InsecureDefaultsService();
+
+    private ObjectName objectName;
+
+    private Map<String, Alert> addedAlerts = new HashMap<>();
+
+    public AlertServiceBean(MBeanServer mBeanServer) throws MalformedObjectNameException {
+        notNull(mBeanServer, "mBeanServer may not be null");
+        this.mBeanServer = mBeanServer;
+        objectName = new ObjectName(OBJECT_NAME);
+    }
+
+    public void init() {
+        registerMBean();
+        LOGGER.debug("Registered [{}] MBean under object name: [{}].",
+                CLASS_NAME,
+                objectName.toString());
+    }
+
+    public void destroy() {
+        if (objectName != null && mBeanServer != null) {
+            unregisterMBean();
+            LOGGER.debug("Unregistered [{}] MBean.", CLASS_NAME);
+        }
+    }
+
+    @Override
+    @Nonnull
+    public List<Map<String, Object>> getAlerts() {
+        List<Map<String, Object>> currentAlerts = new ArrayList<>();
+
+        // add InsecureDefaultsService alerts
+        List<String> insecureDefaultsAlertDetails = insecureDefaultsService.validate()
+                .stream()
+                .map(org.codice.ddf.admin.insecure.defaults.service.Alert::getMessage)
+                .collect(Collectors.toList());
+        if (!insecureDefaultsAlertDetails.isEmpty()) {
+            final Alert insecureDefaultsAlert = new Alert(Alert.Level.ERROR,
+                    INSECURE_DEFAULTS_ALERT_TITLE,
+                    insecureDefaultsAlertDetails);
+            currentAlerts.add(insecureDefaultsAlert);
+        }
+
+        // add alerts from the EventService
+        currentAlerts.addAll(addedAlerts.entrySet()
+                .stream()
+                .map(alertEntry -> {
+                    final Alert alert = alertEntry.getValue();
+                    alert.put("key", alertEntry.getKey());
+                    return alert;
+                })
+                .collect(Collectors.toList()));
+
+        return currentAlerts;
+    }
+
+    @Override
+    @Nonnull
+    public Boolean dismissAlert(@Nonnull String key) {
+        final Alert dismissedAlert = addedAlerts.remove(key);
+        if (dismissedAlert == null) {
+            LOGGER.debug("Failed to dismiss alert with key {}", key);
+            return false;
+        } else {
+            LOGGER.debug("Dismissed alert with title=\"{}\" and key=\"{}\"",
+                    dismissedAlert.getTitle(),
+                    key);
+            return true;
+        }
+    }
+
+    /**
+     * Adds an {@link Alert} to the collection of current {@link Alert}s. The even must have topic
+     * {@link Alert#ALERT_EVENT_TOPIC_VALUE} and
+     * event.getProperty({@link Alert#ALERT_EVENT_ALERT_PROPERTY}) is the {@link Alert} to add.
+     * Already-added {@link Alert}s with the same {@value Alert#KEY} as the {@link Alert} in the event will be
+     * overriden.
```

Comment:
Yes. For example, the AdminAlertAppender adds an alert with a specific key so that it may overwrite the alert that has already been added. The AdminAlertAppender only knows how to push alerts, and we wouldn't want many of the same alert banners to show up. We only want to see one. Instead of checking if an alert with all of the same values already exists in the current alerts, I can specify that I want to overwrite the alert with this specific key if there is one already.